base_layer/core/src/transactions/transaction_components/transaction_output.rs

```diff
@@ -825,6 +840,32 @@ mod test {
         }
     }
 
+    #[test]
+    fn revealed_value_proofs_only_succeed_with_valid_metadata_signatures() {
+        let factories = CryptoFactories::default();
+        let test_params = TestParams::new();
+        let mut output = create_output(
+            &test_params,
+            &factories,
+            MicroTari(20),
+            MicroTari(20),
+            RangeProofType::RevealedValue,
+        )
+        .unwrap();
+        assert!(output.verify_metadata_signature().is_ok());
+        assert!(output.revealed_value_range_proof_check().is_ok());
+
+        output.features.maturity += 1;
```

Comment:
Can this overflow and fail the test? I know we've seen this elsewhere in tests.

Reply:
For the test, it does not matter if it overflows, as long as it is changed, so that the features committed to in the metadata signature is different.

Reply:
Right, I had been concerned that an overflow could panic and intermittent fail the test, depending on how the value was originally set. Looks like it's originally set to zero, so this can't happen. I'd do `output.features.maturity = 1` instead anyway; seems more elegant!